### PR TITLE
fix default release build with VS2015 UP2 compiler warnings

### PR DESCRIPTION
**Pull request by ajax16384**
this will allow to successfully build:
bjam toolset=msvc-14.0 release deprecated-functions=off character-set=unicode link=static warnings-as-errors=on

### CODE REVIEW

File: src/pe_crypto.cpp [hunk 1]
```diff
@@ -187,7 +187,13 @@ namespace libtorrent
 #endif
 
 		int next_barrier = 0;
-		if (iovec.empty() || (next_barrier = m_send_barriers.front().enc_handler->encrypt(iovec)))
+		bool flag = iovec.empty();
```
Comment (arvidn) [inline src/pe_crypto.cpp]: I'm not sure "flag" is a great name here.

File: src/peer_class.cpp [hunk 2]
```diff
@@ -87,7 +87,8 @@ namespace libtorrent
 		}
 		else
 		{
-			ret = m_peer_classes.size();
+			TORRENT_ASSERT(m_peer_classes.size() < 0x100);
+			ret = (peer_class_t)m_peer_classes.size();
```
Comment (arvidn) [inline src/peer_class.cpp]: please use a C++ -style cast instead. I have enabled clang to warn about old-style casts

File: src/session_impl.cpp [hunk 3]
```diff
@@ -919,7 +919,7 @@ namespace aux {
 			TORRENT_ASSERT(e->first.size() <= max_dht_query_length);
 			if (e->first.size() > max_dht_query_length) continue;
 			extension_dht_query registration;
-			registration.query_len = e->first.size();
+			registration.query_len = (uint8_t)e->first.size();
```
Comment (arvidn) [inline src/session_impl.cpp]: please use either static_cast<> or the uint8_t(...) form

File: src/socket_type.cpp [hunk 4]
```diff
@@ -72,19 +72,19 @@ namespace libtorrent
 
 	bool is_utp(socket_type const& s)
 	{
-		return s.get<utp_stream>()
+		return (s.get<utp_stream>() != NULL)
```
Comment (arvidn) [inline src/socket_type.cpp]: the parentheses here seem perhaps a bit excessive (and around the second term too)

File: src/socks5_stream.cpp [hunk 5]
```diff
@@ -194,9 +194,11 @@ namespace libtorrent
 			m_buffer.resize(m_user.size() + m_password.size() + 3);
 			p = &m_buffer[0];
 			write_uint8(1, p);
-			write_uint8(m_user.size(), p);
+			TORRENT_ASSERT(m_user.size() < 0x100);
+			write_uint8((uint8_t)m_user.size(), p);
```
Comment (arvidn) [inline src/socks5_stream.cpp]: please use the uint8_t(...) form instead.

File: src/socks5_stream.cpp [hunk 6]
```diff
@@ -194,9 +194,11 @@ namespace libtorrent
 			m_buffer.resize(m_user.size() + m_password.size() + 3);
 			p = &m_buffer[0];
 			write_uint8(1, p);
-			write_uint8(m_user.size(), p);
+			TORRENT_ASSERT(m_user.size() < 0x100);
```
Comment (arvidn) [inline src/socks5_stream.cpp]: where does 0x100 come from? I would expect the max length to be 255, or (std::numeric_limit<boost::uint8_t>::max())

File: src/storage.cpp [hunk 7]
```diff
@@ -1630,7 +1630,7 @@ namespace libtorrent
 	bool disk_job_fence::has_fence() const
 	{
 		mutex::scoped_lock l(m_mutex);
-		return m_has_fence;
+		return m_has_fence != NULL;
```
Comment (arvidn) [inline src/storage.cpp]: m_has_fence ins an integer. I think it would make more sense to compare to 0

File: src/torrent_info.cpp [hunk 8]
```diff
@@ -1287,7 +1292,7 @@ namespace libtorrent
 			m_merkle_tree[0].assign(root_hash.string_ptr());
 		}
 
-		m_private = info.dict_find_int_value("private", 0);
+		m_private = (info.dict_find_int_value("private", 0) != 0);
```
Comment (arvidn) [inline src/torrent_info.cpp]: the outer parentheses here seem excessive

File: src/udp_socket.cpp [hunk 9]
```diff
@@ -1183,9 +1183,11 @@ void udp_socket::handshake2(error_code const& e)
 		// start sub-negotiation
 		p = &m_tmp_buf[0];
 		write_uint8(1, p);
-		write_uint8(m_proxy_settings.username.size(), p);
+		TORRENT_ASSERT(m_proxy_settings.username.size() < 0x100);
```
Comment (arvidn) [inline src/udp_socket.cpp]: I just realized this was hex, not decimal :P

File: src/udp_socket.cpp [hunk 10]
```diff
@@ -1183,9 +1183,11 @@ void udp_socket::handshake2(error_code const& e)
...
+		write_uint8((uint8_t)m_proxy_settings.username.size(), p);
```
Comment (arvidn) [inline src/udp_socket.cpp]: please use uint8_t(...) form of the cast instead

**Comment by arvidn** [comment]
thanks! I'm hoping to enable warnings-as-errors=on on appveyor

**Comment by arvidn** [comment]
there are still lots of warnings on appveyor. is it msvc-12 vs. msvc-14 discrepancies?

**Comment by ajax16384** [comment]
these warnings are related to examples and boost. i will try to fix examples with the next(separate) pull request.

**Pull request merged**